core/src/main/java/org/eclipse/hono/config/ClientConfigProperties.java

```diff
@@ -94,6 +101,86 @@ public final void setPort(final int port) {
         }
     }
 
+    /**
+     * Gets the name or literal IP address of the proxy host that the client is configured with.
+     *
+     * @return The host name.
+     */
+    public final String getProxyHost() {
+        return proxyHost;
+    }
+
+    /**
+     * Sets the name or literal IP address of the proxy host that the client should use to connect to the an AMQP host
+     *
+     * @param proxyHost The host name or IP address of the proxy
+     * @throws NullPointerException if proxyHost is {@code null}.
+     */
+    public final void setProxyHost(final String proxyHost) {
+        this.proxyHost = Objects.requireNonNull(proxyHost);
+    }
+
+    /**
+     * Gets the TCP port of the proxy server that this client is configured to use.
+     *
+     * @return The port number.
+     */
+    public final int getProxyPort() {
+        return proxyPort;
+    }
+
+    /**
+     * Sets the TCP port of the proxy server that this client should use.
+     *
+     * @param proxyPort The port number of the proxy
+     * @throws IllegalArgumentException if port is 0.
+     */
+    public final void setProxyPort(final int proxyPort) {
+        if (proxyPort != 0) {
```

Comment:
wouldn't `proxyPort > 0` be an even better check?

Reply:
Should check for max value probably as well (but vert.x might do this anyway). Have you spend any thoughts on that?